pkg/cortex/cortex.go

```diff
@@ -218,6 +218,10 @@ func (t *Cortex) initModule(cfg *Config, m moduleName) error {
 
 // Run starts Cortex running, and blocks until a signal is received.
 func (t *Cortex) Run() error {
+	if t.ingester != nil && t.ingester.Config().FlushMode {
+		// If it is a flush mode ingester, skip running the server.
```

Comment:
also skips serving metrics?
This bit doesn't feel right to me.

Reply:
The flush mode would run as a short-lived job. Do we need metrics to keep track of the flushing?

Reply:
Also, it was easier this way to run it as a job and not try and give a signal. The way it would go it the WAL is read during `cortex.New(...)`. `Run()` returns immediately. Flushing happens during `cortex.Stop()`.

Reply:
We need the metrics please. Metrics are super important to debug when things are not working, and I would prefer having metrics here.